kani-compiler/src/kani_queries/mod.rs

```diff
@@ -26,4 +32,25 @@ impl QueryDb {
     pub fn args(&self) -> &Arguments {
         self.args.as_ref().expect("Arguments have not been initialized")
     }
+
+    /// Return a map with model and intrinsic functions defined in Kani's library.
+    ///
+    /// For `kani_core`, those definitions live in the `core` library.
+    ///
+    /// We cache these definitions to avoid doing the lookup every time it is needed.
+    /// The cache should be invalidated if the `stable_mir` context changes.
```

Comment:
See my comment above. The StableMIR ids are only meaningful inside the run function. Once we leave the run context, StableMIR structures may be destroyed or somewhat invalid

Reply:
Ah okay, this makes sense. Can you update your comments to include this information? I'm envisioning a comment that says that the run context starts at the beginning of `codegen_crate` when we invoke `run()`, and if you use any StableMirIds (which includes the cached ones) outside of this run function, that's invalid.
It's similar to what you already have, but just spelling out 1) what the run context is and 2) why the cache contents depend on it to avoid confusion.